src/Interpreters/Aggregator.h

```diff
@@ -1306,6 +1379,9 @@ class Aggregator final
 
     std::vector<bool> is_aggregate_function_compiled;
 
+    // Limit the max keys of adaptive aggregation. Avoid overflow the value id range.
+    static constexpr auto max_adaptive_aggregating_keys = 4l;
```

Comment:
```suggestion
    static constexpr auto max_adaptive_aggregating_keys = 4;
```

motivation: I came across this line N times and N times I asked myself why it is 41